### PR TITLE
Updates for Testbench, Fuzzer, and SRC testbench run

### PR DESCRIPTION
**Pull request by singalsu**
This pull request consists of patches to make Testbench functional after recent pipeline updates. Due to changes in topology parser also Fuzzer is changed to build it successfully. The SRC run script is changed to work with the updated test topologies set.

The test topologies set is trimmed down to contain all DAI permutations only for volume and pass-through pipelines. The algorithm test pipelines are generated only for single I2S setup that is compatible with UP2 device.

### CODE REVIEW

**Comment by lgirdwood** [comment]
@singalsu CI looks unrelated, can you take a look at the SPDX error in checkpatch

**Comment by singalsu** [comment]
> @singalsu CI looks unrelated, can you take a look at the SPDX error in checkpatch

I wonder if there's a problem in checkpatch with shell scripts. The first line usually defines the shell to execute like #!/bin/sh script but checkpatch expects the licence / copyright line to be the first line.

File: tools/testbench/file.c [hunk 1]
```diff
@@ -409,6 +412,8 @@ static struct comp_dev *file_new(const struct comp_driver *drv,
 		(struct sof_ipc_comp_file *)comp;
 	struct file_comp_data *cd;
 
+	comp_cl_info(&comp_file, "file_new()");
```
Comment (singalsu) [inline tools/testbench/file.c]: I remembered this after submitting. My common sense says this should be &comp_file_dai or &comp_file_host. However the code builds and testbench trace output shows file_new() when the components are instantiated by topology. 

I'm not familiar with the new trace functions so need to check this.

Comment (singalsu) [reply tools/testbench/file.c]: I'll change this into debug_print() to avoid future build fail problems.

**Comment by singalsu** [comment]
> > @singalsu CI looks unrelated, can you take a look at the SPDX error in checkpatch
> 
> I wonder if there's a problem in checkpatch with shell scripts. The first line usually defines the shell to execute like #!/bin/sh script but checkpatch expects the licence / copyright line to be the first line.

The beginning of the offending file looks like this. I don't think this can be improved:

```
#!/bin/bash
# SPDX-License-Identifier: BSD-3-Clause
# Copyright(c) 2018 Intel Corporation. All rights reserved.

```

**Comment by paulstelian97** [comment]
> > > @singalsu CI looks unrelated, can you take a look at the SPDX error in checkpatch
> > 
> > 
> > I wonder if there's a problem in checkpatch with shell scripts. The first line usually defines the shell to execute like #!/bin/sh script but checkpatch expects the licence / copyright line to be the first line.
> 
> The beginning of the offending file looks like this. I don't think this can be improved:
> 
> ```
> #!/bin/bash
> # SPDX-License-Identifier: BSD-3-Clause
> # Copyright(c) 2018 Intel Corporation. All rights reserved.
> ```

Yeah, I guess Checkpatch simply isn't aware of this issue, it should be updated to be able to accept shebangs as the first line and work with the SPDX on the second line in that case.

File: tools/testbench/file.c (hunk 1 above)
Comment (singalsu) [reply tools/testbench/file.c]: This is fixed now as debug_print(). The comp_cl_info() macro did not build fail with &comp_file typo so need to be extra careful with those.

There were no other similar cases. I wanted to have this "trace" to see in future the pipeline walks also for only testbench related components for easier fail debugging.

**Comment by singalsu** [comment]
The BYT XCC build issue is unrelated "FLEXnet Licensing error:-96,7".

**Comment by singalsu** [comment]
@ranj063 Do the proposed changes for topology parsing and Fuzzer look OK?

**Comment by lgirdwood** [comment]
CI failures are for known CI issues.

**Pull request merged**